subsys/mgmt/mcumgr/lib/cmd/img_mgmt/Kconfig

```diff
@@ -12,6 +12,22 @@ menuconfig MCUMGR_CMD_IMG_MGMT
 	  Enables mcumgr handlers for image management
 
 if MCUMGR_CMD_IMG_MGMT
+
+if HEAP_MEM_POOL_SIZE > 0
```

Comment:
Not sure where would be best to put it, because it's more something everyone should see from Kconfig not just release notes, perhaps if this was made into a choice selection and the heap one was hidden if not available but display the warning on the static one. The warning being that with a static buffer, multiple concurrent operations (i.e. from different mcumgr transports) will trash the existing context data

Reply:
Concurrent operation, like DFU, has never been possible as the `img_mgmt_impl_write_image_data` will allocate the context and store it within static pointer variable, so the function can run single operation only.
I do not think that concurrent operation was an intention here, rather to move the quite big structure to non-static RAM when possible.

Reply:
It's kind of weird to be using the `alloc` and `free` words in the function names, though, then, isn't it? The difference in behaviors does concern me, too.